File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.scala
object Solution {
  def removeDuplicates(nums: Array[Int]): Int = {
    if (nums.length <= 1) {
      return nums.length
    }

    var j = 1
    val n = nums.length

    for (i <- 2 until n) {
      if (nums(j - 1) != nums(i)) {
        j += 1
        nums(j) = nums(i)
      }
    }
    j + 1
  }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int j = 1;
        int n = nums.length;

        for(int i=2; i< n; i++){
            if(nums[j-1] != nums[i])
                nums[++j] = nums[i];
        }
        return ++j;
    }
}